src/server/server/dal.py

```diff
@@ -1,284 +1,25 @@
 """
 DAL (Data Access Layer) module
-abstract database scheme and ndb API to communicate with the rest of the package through pure (not ndb) Python objects
+abstract database scheme and datastore API to communicate with the rest of the package through pure (not ndb) Python objects
 """
-from google.appengine.ext import ndb
-import dbentities as db  # pylint: disable=relative-import
-import frontendstructs as struct  # pylint: disable=relative-import
-# problem (to solve) with app engine: server is not seen as a package by GAE
-# so you can't do proper relative import (from . import dbentities) as expected by pylint / pep8
+import datetime
+import httplib2
+import jsonpickle
+from gcloud import datastore
+from environment import GCLOUD_PROJECT, IS_TEST_ENV   # pylint: disable=relative-import
+import frontendstructs as struct   # pylint: disable=relative-import
 
 
-def get_user(email):
-    (user, _) = get_user_and_password(email)
-    return user
+REF_FEATURE_SET = u"ref_feature_set"
+NULL_FEATURE_SET = u"null_feature_set"
 
 
-def get_user_and_password(email):
-    db_user = db.User.get(email)
-    return (_to_user(db_user), db_user.password) if db_user else (None, None)
+def _to_urlsafe(key):
+    return jsonpickle.util.b64encode(jsonpickle.encode(key))
```

Comment:
it's more or less what was done with ndb urlsafe (but the equivalent doesn't exists with gcloud API): this is to encode a datastore key as a string that can be passed in an url.